tests/scenario_tests/test_middleware.py

```diff
@@ -183,3 +212,42 @@ def just_next(next):
 
 def just_next_(next_):
     next_()
+
+
+class LazyListenerStarter(Middleware):
+    lazy_called: bool
+    callback_id: str
+
+    def __init__(self, callback_id: str):
+        self.lazy_called = False
+        self.callback_id = callback_id
+
+    def lazy_listener(self):
+        self.lazy_called = True
+
+    def process(self, *, req: BoltRequest, resp: BoltResponse, next: Callable[[], BoltResponse]) -> Optional[BoltResponse]:
```

Comment:
As you can see, this is not so simple and easy, but for advanced developers who want to take full control (including myself), the enhancement by this PR would be greatly valuable.

Reply:
Wow this does provide full control